### PR TITLE
Add support for documentation page redirects

### PR DESCRIPTION
**Pull request by jhlodin**
## Description

Currently when we move or delete pages in the Trino docs (such as in https://github.com/trinodb/trino/commit/698604c8c89f7c50dcceab36b3efe7f6efb83256) any links to the old pages result in 404s forcing users to find the new location for that information on their own.

There are plugins for Sphinx that add support for redirects with a list of mappings in a configuration file. This PR adds the [sphinx-reredirects](https://documatt.gitlab.io/sphinx-reredirects/) plugin which automatically creates search engine-compliant HTML meta redirect pages based on mappings in `conf.py`.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by electrum — APPROVED** [review]

File: docs/src/main/sphinx/conf.py [hunk 1]
```diff
@@ -138,3 +138,11 @@ def setup(app):
 html_css_files = [
     'trino.css',
 ]
+
+# -- List of page redirects ----------------------------------------------------
+# See https://documatt.gitlab.io/sphinx-reredirects/usage.html for plugin docs
+
+redirects = {
+    "installation/jdbc": "client/jdbc",
+    "installation/cli": "client/cli"
```
Comment (electrum) [inline docs/src/main/sphinx/conf.py:147]: Is it legal to have a comma after the last item?

**Comment by jhlodin** [comment]
> I pushed a new image:
> 
> ```
> docker buildx build --push --platform linux/amd64,linux/arm64 -t ghcr.io/trinodb/build/sphinx:6 docs
> ```
> 
> But it fails when building the docs:
> 
> ```
> SPHINX_IMAGE=ghcr.io/trinodb/build/sphinx:6
> 
> Warning, treated as error:
> the sphinx_reredirects extension does not declare if it is safe for parallel reading, assuming it isn't - please ask the extension author to check and make it explicit
> ```

@electrum I ran into the same error, my conclusion is that this particular 3rd party plugin isn't compliant with our Sphinx plugin requirements. My next step for this PR is to experiment with another plugin that has similar functionality and find one that works, leaving this as a draft until I have one to propose.

**Comment by jhlodin** [comment]
I can dig back in since it would help the docs experience with memsql deprecation.

**Comment by jhlodin** [comment]
Cole ran with this in https://github.com/trinodb/trino/pull/16223/commits/3dd724554d8e2d86b44dc21a71363f80c23bd590 so I'm going to close this PR.

**Pull request closed without merging**